### PR TITLE
Avoid data processing by phpspreadsheet

### LINKED ISSUE #387: 2 issues with reports download
Hi @jlevers ,

We are having some issues downloading reports:

1. Fields on GET_LEDGER_DETAIL_VIEW_DATA are being quoted by amazon, so unless I manually change the report type to `ContentType::CSV`, all the data get with double quotes included
2. For any TAB delimited report with a field content starting with = (equal sign), we get the following error:
`code: 0 - content: string(58) "Worksheet!Y16074 -> Formula Error: Unexpected operator '='"`
The code is interpreting it as a formula and trying to process it which I think is incorrect.

I solved second issue updating
`$sheet = $spreadsheet->getSheet(0)->toArray();`
to
`$sheet = $spreadsheet->getSheet(0)->toArray(null, false);`

in Document.php

For first issue, I updated my own code to force the report type to by CSV

Let me know if you agree with those changes (or at least the solution for the second issue) and I can create a PR

### PR DESCRIPTION
**Pull request by 0xSekar**
@jlevers Updated report type in lib/ReportType.php as seems like that file is not auto generated from Amazon API. If you still want to create a special case on Documents.php instead, let me know and will update.

Fixes #387

### CODE REVIEW

File: lib/Document.php [hunk 1]
```diff
@@ -160,7 +160,7 @@ public function download(?bool $postProcess = true, ?string $encoding = null): s
             case ContentType::XLSX:
                 $spreadsheet = $reader->load($this->tmpFilename);
                 if ($this->contentType !== ContentType::XLSX) {
-                    $sheet = $spreadsheet->getSheet(0)->toArray();
+                    $sheet = $spreadsheet->getSheet(0)->toArray(null, false);
```
Comment (jlevers) [inline lib/Document.php:163]: would you mind just linking to the docs for `toArray` so i understand what's going on here? i just did a quick search of their docs and didn't see any `toArray` parameter documentation

**Comment by jlevers** [comment]
let's do the special case in the `Document` class -- it's a bit neater the way you did it, but it's also slightly misleading for anyone reading the `ReportType` code because that report type actually is not a CSV

**Review by jlevers — CHANGES_REQUESTED** [review]

File: lib/Document.php (hunk 1 above)
Comment (0xSekar) [reply lib/Document.php:163]: You can check toArray parameters at https://phpoffice.github.io/PhpSpreadsheet/classes/PhpOffice-PhpSpreadsheet-Worksheet-Worksheet.html#method_toArray

**Comment by 0xSekar** [comment]
> let's do the special case in the `Document` class -- it's a bit neater the way you did it, but it's also slightly misleading for anyone reading the `ReportType` code because that report type actually is not a CSV

Updated

File: lib/Document.php (hunk 1 above)
Comment (jlevers) [reply lib/Document.php:171]: thank you

**Review by jlevers — CHANGES_REQUESTED** [review]
sorry, i should have been more specific -- i think the correct place to do this is in the `switch` statement in the `download` method, right here:

```php
case ContentType::TAB:
                // right here
                $reader->setEnclosure(chr(8));
```

because we're already messing around with the enclosure char, so if we just leave it as its default (`"`) when the report type is `GET_LEDGER_DETAIL_VIEW_DATA`, that should solve the problem. does that make sense to you too? feel free to push back if not (and sorry for all the back and forth on a relatively simple problem)

**Comment by 0xSekar** [comment]
> sorry, i should have been more specific -- i think the correct place to do this is in the `switch` statement in the `download` method, right here:...

Updated. Thanks for the feedback

**Review by jlevers — APPROVED** [review]
Looks good, thanks!

**Pull request merged**